packages/cadl-rlc-emitter/src/transform/transformResponses.ts

```diff
@@ -0,0 +1,77 @@
+import { HeaderMetadata, ImportKind, OperationResponse, ResponseMetadata } from "@azure-tools/rlc-codegen";
+import { Program } from "@cadl-lang/compiler";
+import { getAllRoutes, HttpOperationResponse } from "@cadl-lang/rest/http";
+
+export function transformToResponseTypes(program: Program, importDetails: Map<ImportKind, Set<string>>): OperationResponse[] {
+    const [routes, _diagnostics] = getAllRoutes(program);
+    const rlcResponses: OperationResponse[] = [];
+    let importedModels = new Set<string>();
+    for (const route of routes) {
+        const operationName = `${route.groupName}_${route.operation.name}`;
+        const rlcOperationUnit: OperationResponse = {
+            operationName,
+            responses: []
+        };
+        for (const resp of route.responses) {
+            // TODO: verify status code
+            const statusCode = resp.statusCode == "*" ? `"default"` : `"${resp.statusCode}"`;
+            const rlcResponseUnit: ResponseMetadata = {
+                statusCode,
+                description: resp.description
+            };
+
+            // transform header
+            transformHeaders(resp, rlcResponseUnit);
+            // transform body
+            transformBody(resp, importedModels, rlcResponseUnit)
+            rlcOperationUnit.responses.push(rlcResponseUnit);
+        }
+        rlcResponses.push(rlcOperationUnit);
+    }
+    if (importedModels.size > 0) {
+        importDetails.set(ImportKind.ResponseOutput, importedModels);
+    }
+    return rlcResponses;
+}
+
+function transformHeaders(from: HttpOperationResponse, to: ResponseMetadata) {
```

Comment:
Is from a reserved word ?

Reply:
I don't think it would cause compile errors, but I agree we should try to use another name. Maybe `response` is a good enough name